### ISSUE TITLE
Ddpg agent: `network_state` and `state_spec` do not have matching structures:   . vs.   ()

### ISSUE THREAD

**Issue by Ryo-not-rio** [tensorflow/agents#580]
I'm trying to create a ddpg agent, but I get the following error
    Traceback (most recent call last):
      File "D:/PycharmProjects/SandP_evolution/reinforce_stock.py", line 118, in <module>
        train_loss = agent.train(experience).loss
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\def_function.py", line 780, in __call__
        result = self._call(*args, **kwds)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\def_function.py", line 823, in _call
        self._initialize(args, kwds, add_initializers_to=initializers)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\def_function.py", line 696, in _initialize
        self._stateful_fn._get_concrete_function_internal_garbage_collected(  # pylint: disable=protected-access
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\function.py", line 2855, in _get_concrete_function_internal_garbage_collected
        graph_function, _, _ = self._maybe_define_function(args, kwargs)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\function.py", line 3213, in _maybe_define_function
        graph_function = self._create_graph_function(args, kwargs)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\function.py", line 3065, in _create_graph_function
        func_graph_module.func_graph_from_py_func(
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\framework\func_graph.py", line 986, in func_graph_from_py_func
        func_outputs = python_func(*func_args, **func_kwargs)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tensorflow\python\eager\def_function.py", line 600, in wrapped_fn
        return weak_wrapped_fn().__wrapped__(*args, **kwds)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\agents\tf_agent.py", line 506, in train
        loss_info = self._train_fn(
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\utils\common.py", line 185, in with_check_resource_vars
        return fn(*fn_args, **fn_kwargs)
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\agents\ddpg\ddpg_agent.py", line 237, in _train
        critic_loss = self.critic_loss(time_steps, actions, next_time_steps,
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\agents\ddpg\ddpg_agent.py", line 297, in critic_loss
        target_actions, _ = self._target_actor_network(
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\networks\network.py", line 397, in __call__
        nest_utils.assert_matching_dtypes_and_inner_shapes(
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\utils\nest_utils.py", line 290, in assert_matching_dtypes_and_inner_shapes
        assert_same_structure(
      File "C:\Users\moter\.conda\envs\tensor\lib\site-packages\tf_agents\utils\nest_utils.py", line 113, in assert_same_structure
        raise exception('{}:\n  {}\nvs.\n  {}'.format(message, str1, str2))
    ValueError: <tf_agents.networks.sequential.Sequential object at 0x000001E779CDC490>: `network_state` and `state_spec` do not have matching structures:
      .
    vs.
      ()

Versions:
    tensorflow==2.3.1
    tf-agents==0.6.0

The problem seems to be that network_state seems to not be initialized correctly when the networks are built. I would love some help on this.

**Comment by Ryo-not-rio** [comment]
In nest_utils.py, a comparison between Tensor("Squeeze_5:0", shape=(64,), dtype=int32) and () happens after certain steps and that seems to be causing the issue. Any ideas why?

**Comment by ebrevdo** [comment]
What version of tf_agents are you using?  Try nightly.  We used to have a bug where we passed `target_critic_network(observation, step_type)` and this would accidentally confuse `Sequential` networks that had a call signature expectign `(observation, state)`.  Now we call `target_critic_network(observation, step_type=step_type)` and `step_type` should properly get dropped from being passed to `Sequential`.
